src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -773,42 +762,37 @@ private byte[] engineDecryptFinal(byte[] input, final int offset, final int leng
   @Override
   protected byte[] engineDoFinal(byte[] bytes, int offset, int length)
       throws IllegalBlockSizeException, BadPaddingException {
-    if (opMode == NATIVE_MODE_ENCRYPT) {
-      final byte[] buf = new byte[engineGetOutputSize(length)];
-      int actualLength;
-      try {
-        actualLength = engineEncryptFinal(bytes, offset, length, buf, 0);
-      } catch (ShortBufferException e) {
-        throw new AssertionError(e);
-      }
-
-      if (actualLength == buf.length) {
-        return buf;
-      } else {
-        // This branch should never happen but is technically allowed by the underlying APIs.
-        // So, we cover it just in case.
-        return Arrays.copyOf(buf, actualLength);
+    final byte[] buf = new byte[engineGetOutputSize(length)];
```

Comment:
Allocation of the output buffer now happens here only in the "ACCP allocates the output buffer" interface. We avoid allocating the output buffer in the other two "caller allocates" interfaces.